src/server/controllers/user.js

```diff
@@ -0,0 +1,63 @@
+const { Prisma } = require("@prisma/client");
+const prisma = require("../utils/prisma");
+const secret = process.env.JWT_SECRET;
+
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+const saltRounds = 11;
+
+const getUsers = async (req, res) => {};
+const registerUser = async (req, res) => {
+  const { username, password } = req.body;
+  const hash = await bcrypt.hash(password, saltRounds);
+  try {
+    const user = await prisma.user.create({
+      data: {
+        username: username,
+        password: hash,
+      },
+    });
+    res.status(201).json({ user: user, status: "success" });
+  } catch (e) {
+    res.status(500).json({ error: e.message });
+  }
+};
+
+const logUser = async (req, res) => {
+  const { username, password } = req.body;
+  const user = await prisma.user.findUnique({
+    where: { username: username },
+    select: { password: true },
+  });
+  const createToken = (payload, secret) => {
+    const token = jwt.sign(payload, secret);
+    return token;
+  };
```

Comment:
You could also use this to create a token on register - that way when a user registers they don't have to log in as well

Reply:
They don't have to, my fetch is taking care of that. If someone register they login as well when registering.

Reply:
That's smart! I couldn't see that in your register function, could you explain to me how you handle it?

Reply:
`    fetch("http://localhost:4000/user/register", opts)
      .then((res) => res.json())
      .then(() => {
        setUsers(users);
        fetch("http://localhost:4000/user/register")
          .then((res) => res.json())
          .then((data) => setUsers(data));
        fetch("http://localhost:4000/user/login", opts)
          .then((res) => res.json())
          .then((data) => {
            setLoggedIn(true);
            localStorage.setItem("token", data.data.token);
            localStorage.setItem("username", data.data.username);
            fetch("http://localhost:4000/movie")
              .then((res) => res.json())
              .then((data) => {
                navigate(`/main`);
                setMovies(data.movies);
              });
          });
      });`
      My third fetch is for the login and I use the same opts for both register and login. It doesn't look pretty but it works.

Reply:
My third fetch is for the login and I use the same opts for both register and login. It doesn't look pretty but it works.